test/tests/pbs_nodes_queues.py

```diff
@@ -0,0 +1,58 @@
+# coding utf-8
+
+from ptl.utils.pbs_testsuite import *
+
+
+class TestNodesQueues(PBSTestSuite):
+
+    def setUp(self):
+        PBSTestSuite.setUp(self)
+        self.server.add_resource('foo', 'string', 'h')
+
+        a = {'resources_available.ncpus': 4}
+        self.server.create_vnodes(
+            'vnode', a, 8, self.mom, attrfunc=self.cust_attr)
+
+        a = {'queue_type': 'execution', 'started': 't', 'enabled': 't'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='workq2')
+
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            'queue': 'workq2'}, id='vnode[0]')
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            'queue': 'workq2'}, id='vnode[1]')
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            'queue': 'workq2'}, id='vnode[2]')
+        self.server.manager(MGR_CMD_SET, NODE, {
+                            'queue': 'workq2'}, id='vnode[3]')
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'node_group_key': 'foo'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'node_group_enable': 't'})
+
+    def cust_attr(self, name, totnodes, numnode, attrib):
+        a = {}
+        if numnode % 2 == 0:
+            a['resources_available.foo'] = 'A'
+        else:
+            a['resources_available.foo'] = 'B'
+        return dict(attrib.items() + a.items())
+
+    def tearDown(self):
+        PBSTestSuite.tearDown(self)
+        self.server.manager(MGR_CMD_UNSET, NODE, 'queue', id='vnode[0]')
+        self.server.manager(MGR_CMD_UNSET, NODE, 'queue', id='vnode[1]')
+        self.server.manager(MGR_CMD_UNSET, NODE, 'queue', id='vnode[2]')
+        self.server.manager(MGR_CMD_UNSET, NODE, 'queue', id='vnode[3]')
+
+    def test_pbs7100(self):
+        """
+        Issue with node grouping and nodes associated with queues.  If
+        node_grouping is set at the server level, node/queue association is
+        not honored
+        """
+
+       a = {'Resource_List.select': '2:ncpus=1', 'Resource_List.place': 'vscatter', 'queue': 'workq2'}
```

Comment:
At line 53. It looks like "a =" is preceded by seven spaces rather than eight. Easier to see it in context.